### PR TITLE
Draft: separate agents

### PR DESCRIPTION
**Pull request by wangyoucao577**
### Typical usage     
     
```bash
# build with default agents
task build

# build with minimax_v2v agents
task build AGENT=examples/minimax_v2v

# clean
task clean

# run server 
task server:run
```    

`make build/clean/run-server` still remains for compatibility, but recommend to use `task build/clean/server:run` instead.

### CODE REVIEW

File: agents/Taskfile.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+version: '3'
+
+tasks:
+  build:
```
Comment (plutoless) [inline agents/Taskfile.yml:4]: maybe it's better to call it "use agent" which set corresponding folder as main workspace configs?

File: Taskfile.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    taskfile: ./agents/Taskfile.yml
+    dir: ./agents
+
+  server:
+    taskfile: ./server/Taskfile.yml
+    dir: ./server
+
+tasks:
+  build:
+    desc: build all
```
Comment (plutoless) [inline Taskfile.yml:14]: is this only useful in cicd?

Comment (wangyoucao577) [reply Taskfile.yml:14]: It's equal to `make build`.

File: agents/Taskfile.yml (hunk 1 above)
Comment (wangyoucao577) [reply agents/Taskfile.yml:4]: `task build USE_AGENT=agents/example/minimax_v2v` something like this?

Comment (plutoless) [reply agents/Taskfile.yml:4]: no i mean `task use agents/example/minimax_v2v`

File: Taskfile.yml (hunk 2 above)
Comment (plutoless) [reply Taskfile.yml:14]: but if one extension exists in multiple folders, will it be installed for multiple times?

**Comment by wangyoucao577** [comment]
Replaced by https://github.com/TEN-framework/TEN-Agent/pull/433

**Pull request closed without merging**